src/inheritAttributes.js

```diff
@@ -441,6 +444,39 @@ export const toAdaptationSets = (mpdAttributes, mpdBaseUrls) => (period, index)
   return flatten(adaptationSets.map(toRepresentations(periodAttributes, periodBaseUrls, periodSegmentInfo)));
 };
 
+/**
+ * Tranforms an array of content steering nodes into an object
+ * containing CDN content steering information from the MPD manifest.
+ *
+ * For more information on the DASH spec for Content Steering parsing, see:
+ * https://dashif.org/docs/DASH-IF-CTS-00XX-Content-Steering-Community-Review.pdf
+ *
+ * @param {Node[]} contentSteeringNodes
+ *        Content steering nodes
+ * @return {Object}
+ *        Object containing content steering data
+ */
+export const generateContentSteeringInformation = (contentSteeringNodes) => {
+  // If there are more than one ContentSteering tags, throw an error
+  if (contentSteeringNodes.length > 1) {
+    throw new Error(errors.INVALID_NUMBER_OF_CONTENT_STEERING);
```

Comment:
Curious, is it something that spec obliges?
Can we be less conservative about it, log a warning message, and pick the first one available?

Reply:
Yeah this is a spec requirement. I'm on the fence about this, as it doesn't necessarily prohibit us from achieving playback if they had more than 1 `<ContentSteering>` tag, however the logic simply will not work with more than one available upstream. So I think either error out or pick an arbitrary one. IMO I think it's a little bit more predictable if we just error out, but I could be persuaded otherwise.

Reply:
That makes sense. There isn't really a mechanism for warnings in this repo as there is in others. Would a `console.warn` suffice? Or is there some other way we should go about a warning? I see we often use `videojs.log`, but I am not sure if it is worth adding VJS as a dependency just for this

Reply:
We can use the m3u8 parser approach - to trigger the `warn` event, so there is no need to add VJS or a console.

If you think that this is a completely bad idea to choose the first available node - just ignore me. I didn't read 100% of the spec, so I am pretty sure I have less context here.

Here is my IMHO:
We take whatever is possible from the content and play it.
We notify clients that several features may not work because of specific issues we found.
We do not break playback because some content has some broken features, but is still playable (instead we might want to ignore this feature or define a different approach depending on the situation)

This is what the m3u8 parser does and what browsers do when parsing unpredictable HTML.

Reply:
Good call on using the `warn` event. I think this is the correct approach. For reference, https://github.com/videojs/m3u8-parser/blob/b2d44f204264ff13ede4810ea80fa03156010e26/src/parser.js#L787.

Reply:
Seems like a good idea, however, it looks like that `trigger` function is tied to the `Stream` class in [vjs-utils](https://github.com/videojs/vhs-utils/blob/88d6e10c631e57a5af02c5a62bc7376cd456b4f5/src/stream.js#L62), and that is never used in the MPD parser :/ Ideally I think we could use `Stream` in the MPD parser, but that is a larger change unrelated to this task. My other idea is still just to use `console.warn`, but is it possible I am missing something that would make that simpler?

Reply:
Oh wow, this is a bummer...

Indeed m3u8-parser supports streaming parsing, and mpd parser does not.

And it seems like they have a completely different API:

m3u8:

```javascript
  const parser = new M3u8Parser();

  if (onwarn) {
    parser.on('warn', onwarn);
  }
  if (oninfo) {
    parser.on('info', oninfo);
  }

  customTagParsers.forEach(customParser => parser.addParser(customParser));
  customTagMappers.forEach(mapper => parser.addTagMapper(mapper));

  parser.push(manifestString);
  parser.end();

  return parser.manifest;
```

mpd:
```javascript
// ...
  const manifest = parseMpd(mainXml, {
    manifestUri: srcUrl,
    clientOffset,
    sidxMapping,
    previousManifest
  });

return manifest;
```

> Ideally I think we could use Stream in the MPD parser, but that is a larger change unrelated to this task.
Yes, we would like to support streaming parsing (e.g., for cases when we would like to pipe response stream from fetch to parser, etc...)

Probably, we can use `console.warn` in this case, but we should leave a comment that it should be removed in favor of triggering warn event, once streaming parsing will be implemented

Reply:
There are 2 problems I see with `console.warn`, though:

1. It might be removed if someone processes the parsers script via modification (terser, uglify, etc...)
2. We do not expose any API to enable/disable logging, so it will be logged anyway

Reply:
So we may consider providing a `warn` callback  :)

Reply:
Talked to you a bit offline, and created the logic to handle an `eventHandler` object in this commit: https://github.com/videojs/mpd-parser/pull/174/commits/f6237d79e19c0a569f5af30b2993201b887d3ebd